includes/class-wc-payments-account.php

```diff
@@ -320,6 +320,18 @@ public function maybe_handle_oauth() {
 			try {
 				$this->redirect_to_login();
 			} catch ( Exception $e ) {
+				// Handle the error within the WooCommerce Admin-powered app.
+				if ( self::is_overview_page() ) {
+					wp_safe_redirect(
```

Comment:
For future reference one other way would have been:

https://github.com/Automattic/woocommerce-payments/blob/aabf276642607ea26491dac05babadf29927b63a/includes/class-wc-payments-account.php#L238

https://github.com/Automattic/woocommerce-payments/blob/aabf276642607ea26491dac05babadf29927b63a/includes/admin/class-wc-payments-admin.php#L253